flow360/component/simulation/primitives.py

```diff
@@ -263,16 +263,25 @@ class Box(MultiConstructorBaseModel, _VolumeEntityBase):
     ====
     """
 
+    # private_attribute_input_cache has to be defined first for field validators to work.
+    private_attribute_input_cache: BoxCache = pd.Field(BoxCache(), frozen=True)
```

Comment:
oh, interesting. Question how we can modify type hints..

Reply:
Never mind, I just tested and it seems that even though fields are validated by definition ordering but it is still accessible and has definition (None).

I thought it is like C++ where you reference an uninitialized pointer.

reference: https://docs.pydantic.dev/latest/concepts/validators/#field-validators:~:text=Validation%20is%20done%20in%20the%20order%20fields%20are%20defined%2C%20so%20you%20have%20to%20be%20careful%20when%20using%20ValidationInfo.data%20to%20not%20access%20a%20field%20that%20has%20not%20yet%20been%20validated/populated